homeassistant/components/homekit/security_systems.py

```diff
@@ -0,0 +1,88 @@
+"""Class to hold all alarm control panel accessories."""
+import logging
+
+from homeassistant.helpers.event import async_track_state_change
+
+from . import TYPES
+from .accessories import HomeAccessory
+from .const import (SERV_SECURITY_SYSTEM, CHAR_CURRENT_SECURITY_STATE,
+                    CHAR_TARGET_SECURITY_STATE)
+from homeassistant.const import (STATE_ALARM_ARMED_AWAY,
+                                 STATE_ALARM_ARMED_HOME,
+                                 STATE_ALARM_ARMED_NIGHT,
+                                 STATE_ALARM_DISARMED)
+
+_LOGGER = logging.getLogger(__name__)
+
+HASS_TO_HOMEKIT = {STATE_ALARM_DISARMED: 3, STATE_ALARM_ARMED_HOME: 0,
+                   STATE_ALARM_ARMED_AWAY: 1, STATE_ALARM_ARMED_NIGHT: 2}
+HOMEKIT_TO_HASS = {c: s for s, c in HASS_TO_HOMEKIT.items()}
+STATE_TO_SERVICE = {STATE_ALARM_DISARMED: 'alarm_disarm',
+                    STATE_ALARM_ARMED_HOME: 'alarm_arm_home',
+                    STATE_ALARM_ARMED_AWAY: 'alarm_arm_away',
+                    STATE_ALARM_ARMED_NIGHT: 'alarm_arm_night'}
+
+
+@TYPES.register('SecuritySystem')
+class SecuritySystem(HomeAccessory):
+    """Generate an SecuritySystem accessory for an alarm control panel."""
+
+    def __init__(self, hass, entity_id, display_name):
+        """Initialize a SecuritySystem accessory object."""
+        super().__init__(display_name, entity_id, 'ALARM_SYSTEM')
+
+        self._hass = hass
+        self._entity_id = entity_id
+
+        self.current_security_state = None
+        self.homekit_target_security_state = None
+
+        self.service_alarm = self.get_service(SERV_SECURITY_SYSTEM)
+        self.char_current_state = self.service_alarm.\
+            get_characteristic(CHAR_CURRENT_SECURITY_STATE)
+        self.char_target_state = self.service_alarm.\
+            get_characteristic(CHAR_TARGET_SECURITY_STATE)
+
+        self.char_target_state.setter_callback = self.set_security_state
+
+    def run(self):
+        """Method called be object after driver is started."""
+        state = self._hass.states.get(self._entity_id)
+        self.update_security_state(new_state=state)
+
+        async_track_state_change(self._hass, self._entity_id,
+                                 self.update_security_state)
+
+    def set_security_state(self, value):
+        """Move security state to value if call came from homekit."""
+        if value != self.current_security_state and value in HOMEKIT_TO_HASS:
+            _LOGGER.debug("%s: Set security state to %d",
+                          self._entity_id, value)
+            self.homekit_target_security_state = value
+            hass_value = HOMEKIT_TO_HASS[value]
+            service = STATE_TO_SERVICE[hass_value]
+
+            self._hass.services.call('alarm_control_panel', service,
+                                     {'entity_id': self._entity_id})
```

Comment:
Hmm, I will have to think about this :). In my case, I have an alarm without code in home assistant and I have to admit I have overseen this loss of generality. I will try to have a look at it ASAP

Reply:
I don't have much time at the moment, but maybe plan that the alarm code will be part of the constructor with: alarm_code=None. I will explain later.

Reply:
The current implementation for HomeKit has a huge bug. If you have multiple accessories, it might happen that after a restart of Home Assistant some stop working an only show `Unavailable`. This is due to the way pyhap generates accessory ids. If after a restart those change they will not work.
I plan to add a config option this weekend so that the aid can be set/stored and doesn't change anymore. It would be quiet easy to add an additional parameter for `alarm_panels` which could be used to initialize the accessory. Until then it will only work if someone doesn't have a alarm_code set.

Reply:
I added the alarm_code parameter defaulting to None to the constructor as requested. When it is set, it is passed in the service call to change the alarm state.
Regarding the HomeKit bug: I saw it happening before that accessories stop working after reboot (blocked in the home app), so this explains why..